frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -674,7 +674,10 @@ export const insightLogic = kea<insightLogicType>([
             dashboardsModel.actions.updateDashboardInsight(savedInsight)
 
             const mountedInsightSceneLogic = insightSceneLogic.findMounted()
-            if (redirectToViewMode) {
+            const redirectionPath = decodeParams(router.values.location.search, '?')
```

Comment:
This looks like an open redirect vulnerability. I suggest narrowing this mechanism down to just dashboards. We have a `highlightInsightId` search param in `dashboardLogic`, we could use a similar `originDashboardId` param here. Then we'd call `urls.dashboard(...)` here, avoiding free-form redirects (which can be used maliciously)

Reply:
Got it. Will check.